File: src/main/java/org/lovaprod/model/TransactionHistory.java
package org.lovaprod.model;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class TransactionHistory {
    private String id;
    private List<Transaction> transactions;

    public TransactionHistory() {
        this.id = UUID.randomUUID().toString();
        this.transactions = new ArrayList<>();
    }

    public String getId() {
        return id;
    }

    public List<Transaction> getTransactions() {
        return transactions;
    }

    public void addTransaction(double amount, Transaction.TransactionType type, String destination) {
        Transaction transaction = new Transaction(amount, type, destination);
        transactions.add(transaction);
    }
}

File: TransactionHistory.cs
using System;
using System.Collections.Generic;

namespace WalletApp
{
    public class TransactionHistory
    {
        public string Id { get; }
        public List<Transaction> Transactions { get; }

        public TransactionHistory()
        {
            Id = Guid.NewGuid().ToString();
            Transactions = new List<Transaction>();
        }

        public void AddTransaction(double amount, Transaction.TransactionType type, string destination)
        {
            var transaction = new Transaction(amount, type, destination);
            Transactions.Add(transaction);
        }
    }
}
